### PR TITLE
add org.famistudio.FamiStudio

### PR DESCRIPTION
**Pull request by ernstp**

### CODE REVIEW

**Comment by nedrichards** [comment]
bot, build org.BleuBleu.FamiStudio

**Comment by ernstp** [comment]
bot, build org.BleuBleu.FamiStudio

File: org.BleuBleu.FamiStudio.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+  <id>org.BleuBleu.FamiStudio.desktop</id>
```
Comment (bilelmoussaoui) [inline org.BleuBleu.FamiStudio.appdata.xml:3]:
```suggestion
  <id>org.BleuBleu.FamiStudio</id>
```

Comment (bilelmoussaoui) [reply org.BleuBleu.FamiStudio.appdata.xml:3]: You need a `<launchable type="desktop-id">org.BleuBleu.FamiStudio.desktop</launchable>` as well

File: org.BleuBleu.FamiStudio.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    <content_attribute id="language-humor">none</content_attribute>
+    <content_attribute id="language-discrimination">none</content_attribute>
+    <content_attribute id="social-chat">none</content_attribute>
+    <content_attribute id="social-info">none</content_attribute>
+    <content_attribute id="social-audio">none</content_attribute>
+    <content_attribute id="social-location">none</content_attribute>
+    <content_attribute id="social-contacts">none</content_attribute>
+    <content_attribute id="money-purchasing">none</content_attribute>
+    <content_attribute id="money-gambling">none</content_attribute>
+  </content_rating>
```
Comment (bilelmoussaoui) [inline org.BleuBleu.FamiStudio.appdata.xml:42]:
```suggestion
  <content_rating type="oars-1.1" />
```

File: org.BleuBleu.FamiStudio.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      - type: file
+        path: org.BleuBleu.FamiStudio.svg
+      - type: file
+        path: org.BleuBleu.FamiStudio.desktop
+      - type: file
+        path: org.BleuBleu.FamiStudio.appdata.xml
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=wayland
```
Comment (bilelmoussaoui) [inline org.BleuBleu.FamiStudio.yml:59]: Does it really run under wayland?

File: org.BleuBleu.FamiStudio.yml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  - --socket=pulseaudio
+  - --device=all
+  - --filesystem=home
+  - --env=MONO_GAC_PREFIX=${FLATPAK_DEST}
```
Comment (bilelmoussaoui) [inline org.BleuBleu.FamiStudio.yml:63]: Is this a runtime thing?

File: org.BleuBleu.FamiStudio.yml [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    sources:
+      - type: file
+        path: famistudio.sh
+      - type: file
+        path: org.BleuBleu.FamiStudio.svg
+      - type: file
+        path: org.BleuBleu.FamiStudio.desktop
+      - type: file
+        path: org.BleuBleu.FamiStudio.appdata.xml
+finish-args:
```
Comment (bilelmoussaoui) [inline org.BleuBleu.FamiStudio.yml:56]: Finish-args should be set before the modules by convention.

File: org.BleuBleu.FamiStudio.yml (hunk 3 above)
Comment (ernstp) [reply org.BleuBleu.FamiStudio.yml]: It's using Xwayland of course...

File: org.BleuBleu.FamiStudio.yml (hunk 4 above)
Comment (ernstp) [reply org.BleuBleu.FamiStudio.yml]: Yeah. I had in the wrapper shell script before.

File: org.BleuBleu.FamiStudio.yml [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+      - install -Dm644 org.BleuBleu.FamiStudio.appdata.xml ${FLATPAK_DEST}/share/metainfo/org.BleuBleu.FamiStudio.appdata.xml
+    sources:
+      - type: file
+        path: famistudio.sh
+      - type: file
+        path: org.BleuBleu.FamiStudio.svg
+      - type: file
+        path: org.BleuBleu.FamiStudio.desktop
+      - type: file
+        path: org.BleuBleu.FamiStudio.appdata.xml
```
Comment (LeandroStanger) [inline org.BleuBleu.FamiStudio.yml:65]: New appdata files should use metainfo instead. https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html

**Comment by ernstp** [comment]
https://github.com/BleuBleu who is the upstream author should have write access to the new repository also.

File: org.BleuBleu.FamiStudio.appdata.xml [hunk 7]
```diff
@@ -0,0 +1,42 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (bilelmoussaoui) [inline org.BleuBleu.FamiStudio.appdata.xml:1]: Would be nice to send this upstream if possible.

File: org.BleuBleu.FamiStudio.desktop [hunk 8]
```diff
@@ -0,0 +1,8 @@
+[Desktop Entry]
```
Comment (bilelmoussaoui) [inline org.BleuBleu.FamiStudio.desktop:1]: Same as the metainfo

File: org.BleuBleu.FamiStudio.yml [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+  - --share=ipc
+  - --socket=x11
+  - --socket=pulseaudio
+# Read and write projects
+  - --filesystem=home
+# For the Mono runtime
+  - --env=MONO_GAC_PREFIX=${FLATPAK_DEST}
+# Midi keyboard input
+  - --device=all
+modules:
```
Comment (barthalion) [inline org.BleuBleu.FamiStudio.yml:18]: Not critical but should be reformatted to follow https://github.com/flathub/flathub/wiki/YAML-Style-Guide.

File: org.BleuBleu.FamiStudio.yml [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+    build-commands:
+      - install -d /app/bin/
+      - cp -r * /app/bin/
+      - ln -sf /usr/lib/x86_64-linux-gnu/libopenal.so.1 /app/bin/libopenal32.so
+    sources:
+      - type: archive
+        url: https://github.com/BleuBleu/FamiStudio/releases/download/2.2.0/FamiStudio220-LinuxAMD64.zip
+        sha256: 56e1e1c0a31c61a12b657320d976f5c6b85e6ebb3b0a5b38a36460d583e4bf9f
+        strip-components: 0
+  - name: extrafiles
```
Comment (barthalion) [inline org.BleuBleu.FamiStudio.yml:50]: Any reason to do it in separate module and not `post-install` of the app itself?

**Comment by barthalion** [comment]
bot, build org.BleuBleu.FamiStudio

File: org.BleuBleu.FamiStudio.appdata.xml [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+<component type="desktop">
+  <id>org.BleuBleu.FamiStudio</id>
```
Comment (barthalion) [inline org.BleuBleu.FamiStudio.appdata.xml:3]: This should be actual project domain, or example `org.famistudio.FamiStudio`.

**Review by barthalion — CHANGES_REQUESTED** [review]

File: org.BleuBleu.FamiStudio.yml (hunk 9 above)
Comment (ernstp) [reply org.BleuBleu.FamiStudio.yml:18]: Sorry, what do you want me to change exactly?

Comment (barthalion) [reply org.BleuBleu.FamiStudio.yml:18]: Modules should be separated by new lines, mostly.

**Comment by barthalion** [comment]
bot, build org.famistudio.FamiStudio

**Comment by ernstp** [comment]
Only changed the icon to a better SVG with all the details.

**Pull request closed without merging**